### PR TITLE
Add facetting support for the v2.

### PR DESCRIPTION
**Pull request by phndiaye**
### What does this PR do?

Related to : #

### What are the observable changes?

Screenshot 2022-02-02 at 11 45 33
Screenshot 2022-02-02 at 11 46 09
Screenshot 2022-02-02 at 11 47 04

### 🧑‍💻 Developer Heads Up

⚡ Since we are using [Ember Octane](https://blog.emberjs.com/octane-is-here/) now:
* Feel free to migrate existing components to Glimmer Components.
* Write new ones exclusively in it.

Useful Resource : [Ember Octane vs Classic Cheat Sheet](https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/)

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/hyper-table-v2/facets-loader.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        facetFilter = { key: this.args.filteringKey, value: '' };
+      }
+
+      this.args.handler.applyFilters(this.args.column, [facetFilter]).then(() => {
+        this.appliedFacets = this.appliedFacets.filter((x) => x !== facet.identifier);
+      });
+    }
+  }
+
+  private fetchFacets() {
```
Comment (JulienVannier66) [inline addon/components/hyper-table-v2/facets-loader.ts:85]:
```suggestion
  private fetchFacets(): void {
```

**Comment by JulienVannier66** [comment]
@phndiaye It is possible to move faceting into `addon/components/hyper-table-v2/filtering-renderers/common` with other common component created by @Miexil. What you you think?

**Review by JulienVannier66 — APPROVED** [review]
Otherwise LGTM

File: addon/core/handler.ts [hunk 2]
```diff
@@ -312,7 +317,24 @@ export default class TableHandler {
...
-    this.columns = columns;
+
+    let shouldRedraw = false;
+
+    columns.forEach((column) => {
+      let existingColumn = this.columns.find((c) => c.definition.key === column.definition.key);
+
+      if (existingColumn) {
+        existingColumn = column;
+      } else {
```
Comment (Miexil) [inline addon/core/handler.ts:328]: I'm not sure what `existingColumn` is used for ? you assign it to `column` if it is non-null but you don't use it anywhere else in the function's scope ? 🤔

Comment (Miexil) [reply addon/core/handler.ts:328]: All good !

**Review by Miexil — APPROVED** [review]

**Pull request merged**